### ISSUE TITLE
"name='NID', domain=None, path=None"

### ISSUE THREAD

**Issue by 4nn4r3ll4** [vaibhavk97/GoBooDo#22]
I started downloading a book and everything was going fine than it looks like I had some connection issue and when I tried resuming the process I got the following error and I'm stuck with it:
```
 .88888.            888888ba                    888888ba           
d8'   `88           88    `8b                   88    `8b          
88        .d8888b. a88aaaa8P' .d8888b. .d8888b. 88     88 .d8888b. 
88   YP88 88'  `88  88   `8b. 88'  `88 88'  `88 88     88 88'  `88 
Y8.   .88 88.  .88  88    .88 88.  .88 88.  .88 88    .8P 88.  .88 
 `88888'  `88888P'  88888888P `88888P' `88888P' 8888888P  `88888P' 
ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo                                                                                                                                
    
"name='NID', domain=None, path=None"
Received invalid response
```
How can I solve this? Thanks very much in advance :)

**Comment by 4nn4r3ll4** [comment]
Maybe it's worth mentioning that I left proxies.txt unchanged and that this is my settings.json:
```
{
    "country": "it",
    "page_resolution": 1500,
    "tesseract_path": "/usr/bin/tesseract",
    "proxy_links": 0,
    "proxy_images": 0,
    "max_retry_links": 1,
    "max_retry_images": 1,
    "global_retry_time":3
}
```

**Comment by 4nn4r3ll4** [comment]
Nevermind, all solved. Thanks for the great tool!

**Comment by athossampayo** [comment]
okay so tried debugging this and seems to be an error that happens when Google redirects you to the captcha page because the proxies are detected as suspicious ip's. When checking the real content of the request.get() in line 50 of GoBooDo.py it gave me:

`b'<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">\n<html>\n<head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta name="viewport" content="initial-scale=1"><title>https://google.com.br/</title></head>\n<body style="font-family: arial, sans-serif; background-color: #fff; color: #000; padding:20px; font-size:18px;" onload="e=document.getElementById(\'captcha\');if(e){e.focus();}">\n<div style="max-width:400px;">\n<hr noshade size="1" style="color:#ccc; background-color:#ccc;"><br>\n<form id="captcha-form" action="index" method="post">\n<script src="https://www.google.com/recaptcha/api.js" async defer></script>\n<script>var submitCallback = function(response) {document.getElementById(\'captcha-form\').submit();};</script>\n<div id="recaptcha" class="g-recaptcha" data-sitekey="6LfwuyUTAAAAAOAmoS0fdqijC2PbbdH4kjq62Y1b" data-callback="submitCallback" data-s="FCd7_Dkh8CGC6q2mZ-lk0nBIbXQfvZ6OxMMcA71lEHCkYgU5J9KDH1LzP4ZWkX-zkY6cMHWpTt-CAoIUScyf277jRPoffOmNoRlbUc29LnAHL_Sd8nMcrJriAE4B_xR_0_DfFhLvgTKdk9eO1BXrSRYklI7S3n2sZ0i4F9MQVIXTYubrtc3ggdzvfLTDmkTedbIETQpLxJsaFf87SzZLOPjpFgA5doQMnx-wramCLj336VjW4ohP4nI"></div>\n<input type=\'hidden\' name=\'q\' value=\'EhAoBAFMAWaI15XQhJ2RtZIpGJGzp_YFIhkA8aeDS4KECAX5AUcKunwfcItq3kfuw-eNMgFy\'><input type="hidden" name="continue" value="https://google.com.br/">\n</form>\n<hr noshade size="1" style="color:#ccc; background-color:#ccc;">\n\n<div style="font-size:13px;">\n<b>About this page</b><br><br>\n\nOur systems have detected unusual traffic from your computer network.  This page checks to see if it&#39;s really you sending the requests, and not a robot.  <a href="#" onclick="document.getElementById(\'infoDiv\').style.display=\'block\';">Why did this happen?</a><br><br>\n\n<div id="infoDiv" style="display:none; background-color:#eee; padding:10px; margin:0 0 15px 0; line-height:1.4em;">\nThis page appears when Google automatically detects requests coming from your computer network which appear to be in violation of the <a href="//www.google.com/policies/terms/">Terms of Service</a>. The block will expire shortly after those requests stop.  In the meantime, solving the above CAPTCHA will let you continue to use our services.<br><br>This traffic may have been sent by malicious software, a browser plug-in, or a script that sends automated requests.  If you share your network connection, ask your administrator for help &mdash; a different computer using the same IP address may be responsible.  <a href="//support.google.com/websearch/answer/86640">Learn more</a><br><br>Sometimes you may be asked to solve the CAPTCHA if you are using advanced terms that robots are known to use, or sending requests very quickly.\n</div>\n\nIP address: <br>Time: 2020-05-24T02:18:58Z<br>URL: https://google.com.br/<br>\n</div>\n</div>\n</body>\n</html>\n'`

**Comment by vaibhavk97** [comment]
Great find. During my testing of the program, I never encountered this issue.

**Comment by athossampayo** [comment]
> Great find. During my testing of the program, I never encountered this issue.

If you want I can try to pull a request to add some detection to it and print an message so the user isn't confused.

**Comment by vaibhavk97** [comment]
> > Great find. During my testing of the program, I never encountered this issue.
> 
> If you want I can try to pull a request to add some detection to it and print an message so the user isn't confused.

Sure, please go ahead.

**Comment by athossampayo** [comment]
> > > Great find. During my testing of the program, I never encountered this issue.
> > 
> > 
> > If you want I can try to pull a request to add some detection to it and print an message so the user isn't confused.
> 
> Sure, please go ahead.

What I found is that the `resethead()` function is running without the proxies (the proxies are chosen after it run) and after some checks google blocks the user IP because too much `resethead()` requests . The thing is that it might be circunvented by defining the proxy before and using it in this function too. I tried it but don't know why got timeout every try. For now I will do the pull request with the print saying to the user to wait some time before trying again, so google can relax with its IP, but will try to find a way to use the proxies before this first request. Sorry if got too confused, english isn't my first language.
